### ISSUE TITLE
compile errors

### ISSUE THREAD

**Issue by jyasuu** [jyasuu/llm-can-schedule#5]
Compiling jssp-scheduler-real v0.1.0 (/mnt/d/git/llm-can-schedule/integrated-real-train)
error[E0061]: this function takes 2 arguments but 3 arguments were supplied
  --> src/model.rs:32:25
   |
32 |         let embedding = Linear::new(
   |                         ^^^^^^^^^^^
33 |             config.vocab_size,
   |             ----------------- expected `Tensor`, found `usize`
34 |             config.hidden_dim,
35 |             vs.pp("embedding"),
   |             ------------------ unexpected argument #3 of type `VarBuilderArgs<'_, Box<dyn SimpleBackend>>`
   |
note: expected `Option<Tensor>`, found `usize`
  --> src/model.rs:34:13
   |
34 |             config.hidden_dim,
   |             ^^^^^^^^^^^^^^^^^
   = note: expected enum `std::option::Option<candle_core::Tensor>`
              found type `usize`
note: associated function defined here
  --> /home/sysadmin/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/candle-nn-0.8.4/src/linear.rs:29:12
   |
29 |     pub fn new(weight: Tensor, bias: Option<Tensor>) -> Self {
   |            ^^^
help: remove the extra argument
   |
33 ~             /* candle_core::Tensor */,
34 ~             /* std::option::Option<candle_core::Tensor> */,
   |

error[E0277]: the `?` operator can only be applied to values that implement `Try`
  --> src/model.rs:32:25
   |
32 |           let embedding = Linear::new(
   |  _________________________^
33 | |             config.vocab_size,
34 | |             config.hidden_dim,
35 | |             vs.pp("embedding"),
36 | |         )?;
   | |__________^ the `?` operator cannot be applied to type `candle_nn::Linear`
   |
   = help: the trait `Try` is not implemented for `candle_nn::Linear`

error[E0308]: arguments to this function are incorrect
  --> src/model.rs:41:25
   |
41 |             let layer = Linear::new(
   |                         ^^^^^^^^^^^
42 |                 config.hidden_dim,
   |                 ----------------- expected `Tensor`, found `usize`
   |
note: expected `Option<Tensor>`, found `VarBuilderArgs<'_, Box<...>>`
  --> src/model.rs:43:17
   |
43 |                 vs.pp(format!("layer_{}", i)),
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: expected enum `std::option::Option<candle_core::Tensor>`
            found struct `VarBuilderArgs<'_, Box<dyn SimpleBackend>>`
note: associated function defined here
  --> /home/sysadmin/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/candle-nn-0.8.4/src/linear.rs:29:12
   |
29 |     pub fn new(weight: Tensor, bias: Option<Tensor>) -> Self {
   |            ^^^

error[E0061]: this function takes 2 arguments but 3 arguments were supplied
  --> src/model.rs:49:27
   |
49 |         let output_head = Linear::new(
   |                           ^^^^^^^^^^^
50 |             config.hidden_dim,
   |             ----------------- expected `Tensor`, found `usize`
51 |             config.vocab_size,
52 |             vs.pp("output"),
   |             --------------- unexpected argument #3 of type `VarBuilderArgs<'_, Box<dyn SimpleBackend>>`
   |
note: expected `Option<Tensor>`, found `usize`
  --> src/model.rs:51:13
   |
51 |             config.vocab_size,
   |             ^^^^^^^^^^^^^^^^^
   = note: expected enum `std::option::Option<candle_core::Tensor>`
              found type `usize`
note: associated function defined here
  --> /home/sysadmin/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/candle-nn-0.8.4/src/linear.rs:29:12
   |
29 |     pub fn new(weight: Tensor, bias: Option<Tensor>) -> Self {
   |            ^^^
help: remove the extra argument
   |
50 ~             /* candle_core::Tensor */,
51 ~             /* std::option::Option<candle_core::Tensor> */,
   |

error[E0599]: no method named `clamp_min` found for struct `candle_core::Tensor` in the current scope
    --> src/model.rs:81:29
     |
81   |             hidden = hidden.clamp_min(&Tensor::new(&[0.0_f32], &self.device)?)?;
     |                             ^^^^^^^^^
     |
help: there is a method `clamp` with a similar name, but with different arguments
    --> /home/sysadmin/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/candle-core-0.8.4/src/tensor.rs:1108:5
     |
1108 |     pub fn clamp<T1: TensorOrScalar, T2: TensorOrScalar>(&self, min: T1, max: T2) -> Result<Self> {
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0599]: no method named `unwrap` found for struct `VarBuilderArgs` in the current scope
   --> src/model.rs:113:62
    |
113 |         let vs = VarBuilder::zeros(DType::F32, &self.device).unwrap();
    |                                                              ^^^^^^ method not found in `VarBuilderArgs<'_, Box<dyn SimpleBackend>>` 

error[E0308]: arguments to this function are incorrect
  --> src/trainer.rs:26:25
   |
26 |         let optimizer = AdamW::new(params, learning_rate)?;
   |                         ^^^^^^^^^^         ------------- expected `ParamsAdamW`, found `f64`
   |
note: expected `Vec<Var>`, found `Vec<Tensor>`
  --> src/trainer.rs:26:36
   |
26 |         let optimizer = AdamW::new(params, learning_rate)?;
   |                                    ^^^^^^
   = note: expected struct `Vec<Var>`
              found struct `Vec<candle_core::Tensor>`
note: associated function defined here
  --> /home/sysadmin/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/candle-nn-0.8.4/src/optim.rs:8:8
   |
8  |     fn new(vars: Vec<Var>, config: Self::Config) -> Result<Self>;
   |        ^^^

error[E0277]: the trait bound `&Vec<i64>: NdArray` is not satisfied
   --> src/trainer.rs:155:13
    |
154 |         let input_ids = Tensor::new(
    |                         ----------- required by a bound introduced by this call
155 |             &flat.iter().map(|&x| x as i64).collect::<Vec<_>>(),
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `WithDType` is not implemented for `&Vec<i64>`
    |
    = note: required for `&Vec<i64>` to implement `NdArray`
note: required by a bound in `candle_core::Tensor::new`
   --> /home/sysadmin/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/candle-core-0.8.4/src/tensor.rs:366:19
    |
366 |     pub fn new<A: crate::device::NdArray>(array: A, device: &Device) -> Result<Self> {
    |                   ^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `Tensor::new`
help: consider dereferencing here
    |
155 |             &*flat.iter().map(|&x| x as i64).collect::<Vec<_>>(),
    |              +

Some errors have detailed explanations: E0061, E0277, E0308, E0599.
For more information about an error, try `rustc --explain E0061`.
error: could not compile `jssp-scheduler-real` (bin "jssp-scheduler") due to 8 previous errors
